### ISSUE TITLE
[Bug]: jei is not showing or working

### ISSUE THREAD

**Issue by sophienix** [mezz/JustEnoughItems#2749]
### Steps to Reproduce the Bug

i have jei in 1.18.2 and it wont work i used curseforge to add it to my modpack but it wont show up i cans fine recipies i even tried useing the show recipy key and it wont show recipies its in the control list but whn i try and use it it wont show any thing

### Expected Behavior

it supose to show up 

### Actual Behavior

its now showing

**Comment by agentwolf88** [comment]
Same

**Comment by sophienix** [comment]
> Same

might because 1.18.2 moded is still in beta as well as jei is also in beta cause the other mods i have works just not jei

**Comment by Coraline77447** [comment]
I'm also experiencing the same issue as well

**Comment by NixSar** [comment]
I have the issue on 1.18.1 as well.

**Comment by Talfereg** [comment]
Yea on single player it works no problem but now it doesn't work on my dedicated server. Very odd. Possible mod conflict, but why would it work in one instance and not another?

**Comment by GWgameing** [comment]
I am having this problem too. We are running it on a small 1.18.1 server  (my friend and I). A temporary fix we have found is to exit and rejoin.

**Comment by mezz** [comment]
Thanks for the report!
I need some more information to solve this issue, please paste your full `fml-client-latest.log` or `latest.log` to https://gist.github.com and link it here.

**Comment by GWgameing** [comment]
Here you go.
[Latest.log](https://gist.github.com/GWgameing/d4379af1930efbff8b34d81f2e37f24d/1b4dfe90ac6ceb929eb74ddc4b17dcae87858802)
Here is also our pack if it helps. 
[New Server Pack.zip](https://github.com/mezz/JustEnoughItems/files/8328617/New.Server.Pack.zip)
P.s. I included the logs in the mod pack.

**Comment by mezz** [comment]
I have released a new version of JEI that should help debug this issue with more logs.
Please try out version `1.18.2-9.5.4.165` and it should fix the bug. 
Even if it's fixed, please paste the latest.log here so I can understand better.

**Comment by GWgameing** [comment]
Ok, is there a way I can backport this to 1.18.1. I did notice this thread was originally intended for 1.18.2, but wanted to also draw your attention to it occurring on 1.18.1. I fully understand if this is not possible, or if the mod has been discontinued on 1.18.1.

**Comment by mezz** [comment]
I don't really want to support 1.18.1 much longer but I understand 1.18.2 is actually a pretty big update for a lot of mods, so I'll see what I can do.

**Comment by mezz** [comment]
I was able to implement similar changes in `9.4.1.168` for 1.18.1

**Comment by mezz** [comment]
Please try this out and paste the new `latest.log` here so I can see what might be causing the original problem.

**Comment by GWgameing** [comment]
It appears to have worked, atleast for 1.18.1. 
[Here is the latest.log](https://gist.github.com/GWgameing/97f92bb1d3f791bf328c9e1308dcae7d)
Thank You!

**Comment by GWgameing** [comment]
Ok so after a little more testing, I have noticed that it will appear without having to disconnect and rejoin, but the recipies will not show. For example, I can search for a furnace, but it will not give me the pop up on how to craft it.
[I know Data helps, so here is another latest.log](https://gist.github.com/GWgameing/00a1f12f7a1ee681253eefa2d72c5ea0)

**Comment by mezz** [comment]
Thanks for the logs!
It looks like `RecipesUpdatedEvent` is not working because a mod is crashing when handling it. This was causing JEI to never start (it depends on the event) and is also causing the issues with recipes not showing.

Here's the relevant crash from your log that causes this problem:
```
[20:30:55] [Render thread/FATAL]: Error executing task on Client
java.lang.IllegalStateException: Tag voidtotem:additional_totems used before it was bound
	at net.minecraft.tags.StaticTagHelper$Wrapper.m_13263_(StaticTagHelper.java:142) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.tags.StaticTagHelper$Wrapper.m_8110_(StaticTagHelper.java:153) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.world.item.ItemStack.m_150922_(ItemStack.java:204) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at com.affehund.voidtotem.core.ModUtils.isAdditionalTotem(ModUtils.java:119) ~[voidtotem-forge-1.18.1-1.2.0.jar%23188!/:1.18.1-1.2.0]
	at com.affehund.voidtotem.core.ModUtils.isVoidTotemOrAdditionalTotem(ModUtils.java:111) ~[voidtotem-forge-1.18.1-1.2.0.jar%23188!/:1.18.1-1.2.0]
	at com.affehund.voidtotem.VoidTotem.attachCaps(VoidTotem.java:96) ~[voidtotem-forge-1.18.1-1.2.0.jar%23188!/:1.18.1-1.2.0]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.event.ForgeEventFactory.gatherCapabilities(ForgeEventFactory.java:576) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraftforge.event.ForgeEventFactory.gatherCapabilities(ForgeEventFactory.java:570) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraftforge.common.capabilities.CapabilityProvider.doGatherCapabilities(CapabilityProvider.java:73) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraftforge.common.capabilities.CapabilityProvider.getCapabilities(CapabilityProvider.java:87) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraftforge.common.capabilities.CapabilityProvider.getCapability(CapabilityProvider.java:182) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraftforge.common.capabilities.ICapabilityProvider.getCapability(ICapabilityProvider.java:34) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at com.maciej916.indreb.common.util.CapabilityUtil.getCapabilityHelper(CapabilityUtil.java:17) ~[indreb-1.18.1-0.12.4.jar%23105!/:1.18.1-0.12.4]
	at com.maciej916.indreb.common.item.base.ElectricItem.readShareTag(ElectricItem.java:135) ~[indreb-1.18.1-0.12.4.jar%23105!/:1.18.1-0.12.4]
	at net.minecraftforge.common.extensions.IForgeItemStack.readShareTag(IForgeItemStack.java:361) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_130267_(FriendlyByteBuf.java:476) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.world.item.crafting.ShapedRecipe$Serializer.m_8005_(ShapedRecipe.java:308) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.world.item.crafting.ShapedRecipe$Serializer.m_8005_(ShapedRecipe.java:285) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateRecipesPacket.m_133647_(ClientboundUpdateRecipesPacket.java:45) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_178371_(FriendlyByteBuf.java:119) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_178366_(FriendlyByteBuf.java:135) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateRecipesPacket.<init>(ClientboundUpdateRecipesPacket.java:22) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.ConnectionProtocol$PacketSet.m_178327_(ConnectionProtocol.java:417) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.ConnectionProtocol.m_178321_(ConnectionProtocol.java:469) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraftforge.network.filters.VanillaPacketSplitter.onClientPacket(VanillaPacketSplitter.java:146) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.network.NetworkInstance.dispatch(NetworkInstance.java:68) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraftforge.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:75) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at net.minecraftforge.network.NetworkHooks.onCustomPayload(NetworkHooks.java:75) ~[forge-1.18.1-39.1.1-universal.jar%23195!/:?]
	at net.minecraft.client.multiplayer.ClientPacketListener.m_7413_(ClientPacketListener.java:1812) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.m_5797_(ClientboundCustomPayloadPacket.java:57) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket.m_5797_(ClientboundCustomPayloadPacket.java:7) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.network.protocol.PacketUtils.m_131356_(PacketUtils.java:21) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:151) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:125) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:110) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1010) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:661) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at net.minecraft.client.main.Main.main(Main.java:205) ~[client-1.18.1-20211210.034407-srg.jar%23191!/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:45) ~[fmlloader-1.18.1-39.1.1.jar%2323!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:149) [bootstraplauncher-1.0.0.jar:?]
```

I am not 100% sure, but it looks like the mod `affehund.voidtotem` is causing this problem. 
Please report this issue and crash log to them here, and link back to this JEI issue so that they have context: https://github.com/Affehund/VoidTotem/issues

**Comment by Charmanth** [comment]
I'm having this issue without the afforementioned mod installed.  The items load in JEI, but base crafting recipes are not.  For example, red wool will load, but not how to craft it in the crafting grid, but it'll show in villager trades, mod drops (from JER) and such.

After some more testing, it seems to be an issue with servers?  It worked fine when I tried it in single player, but my server isn't having it.

Edit:  [GIST of the latest log file.](https://gist.github.com/Charmanth/bc9ad0828e1e3d85a488088265b582bb)

**Comment by mezz** [comment]
@Charmanth if your bug doesn't have the same cause as this one, please open a new issue so we can investigate separately.

Edit: nevermind, this could be the same issue but you have posted a Server log. Please link the client's latest.log here and I'll take a look.

**Comment by Charmanth** [comment]
[client logs](https://gist.github.com/Charmanth/b95a4746a455938e0bc1f023f635d4ab)  Like I said, it works fine with single player.  But it's not working on the server.

**Comment by mezz** [comment]
thanks for the logs!
yeah you're having the same error as the one described above
```
[29Mar2022 18:40:57.034] [Render thread/ERROR] [mezz.jei.startup.StartEventObserver/]: A Screen is opening but JEI hasn't started yet.
Normally, JEI is started after ClientPlayerNetworkEvent.LoggedInEvent, TagsUpdatedEvent, and RecipesUpdatedEvent.
Something has caused one or more of these events to fail, so JEI is starting very late.
```

Looking further up the log for which event is crashing, I see:
```
[29Mar2022 18:40:46.748] [Render thread/FATAL] [net.minecraft.util.thread.BlockableEventLoop/]: Error executing task on Client
net.minecraft.ResourceLocationException: Non [a-z0-9/._-] character in path of location: minecraft:net.minecraft.world.level.material.WaterFluid$Source@4bf89d3d
	at net.minecraft.resources.ResourceLocation.<init>(ResourceLocation.java:37) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at net.minecraft.resources.ResourceLocation.<init>(ResourceLocation.java:42) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at cech12.bucketlib.api.crafting.FluidIngredient$Serializer.parse(FluidIngredient.java:153) ~[bucketlib-1.18.1-0.5.0.1.jar%2369!/:1.18.1-0.5.0.1]
	at cech12.bucketlib.api.crafting.FluidIngredient$Serializer.parse(FluidIngredient.java:138) ~[bucketlib-1.18.1-0.5.0.1.jar%2369!/:1.18.1-0.5.0.1]
	at net.minecraftforge.common.crafting.CraftingHelper.getIngredient(CraftingHelper.java:95) ~[forge-1.18.1-39.1.2-universal.jar%23186!/:?]
	at net.minecraft.world.item.crafting.Ingredient.m_43940_(Ingredient.java:193) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at net.minecraft.world.item.crafting.ShapelessRecipe$Serializer.m_8005_(ShapelessRecipe.java:111) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at net.minecraft.world.item.crafting.ShapelessRecipe$Serializer.m_8005_(ShapelessRecipe.java:77) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateRecipesPacket.m_133647_(ClientboundUpdateRecipesPacket.java:45) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_178371_(FriendlyByteBuf.java:119) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at net.minecraft.network.FriendlyByteBuf.m_178366_(FriendlyByteBuf.java:135) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateRecipesPacket.<init>(ClientboundUpdateRecipesPacket.java:22) ~[client-1.18.1-20211210.034407-srg.jar%23182!/:?]
...
```

This is a crash from `cech12.bucketlib` which is breaking the `ClientboundUpdateRecipesPacket`, such that the recipes coming from the server are not able to be processed on the client.
Please report the issue to them here with your client log: https://github.com/cech12/BucketLib/issues

**Comment by Charmanth** [comment]
Thanks, I'll do that!

**Comment by mezz** [comment]
This has been fixed in BucketLib [1.18.1-0.5.0.2](https://github.com/cech12/BucketLib/releases/tag/1.18.1-0.5.0.2)

**Comment by cech12** [comment]
The issue reported by @Charmanth was on my side and is fixed in the latest version of Bucketlib.
